engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,147 @@
+package osxnative
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+// white list of environment variables that must
+// be passed to the child process
+var environmentWhitelist = []string{
+	"PATH",
+	"HOME",
+	"USER",
+	"SHELL",
+	"TMPDIR",
+	"PWD",
+	"EDITOR",
+	"LANG",
+	"LOGNAME",
+	"TERM",
+	"TERM_PROGRAM",
+}
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func downloadLink(link string) (string, error) {
+	resp, err := http.Get(link)
+
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	contentDisposition := resp.Header.Get("Content-Disposition")
+	_, params, err := mime.ParseMediaType(contentDisposition)
+
+	if err != nil {
+		return "", err
+	}
+
+	filename := params["filename"]
+	file, err := os.Create(filename)
+
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		os.Remove(filename)
+		return "", err
+	}
+
+	return filename, nil
+}
+
+func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
+	if s.aborted {
+		return resultset{success: false}, engines.ErrSandboxAborted
+	}
+
+	if s.taskPayload.Link != "" {
+		filename, err := downloadLink(s.taskPayload.Link)
+
+		if err != nil {
+			s.context.LogError(err)
+			return nil, engines.ErrNonFatalInternalError
+		}
+
+		defer os.Remove(filename)
+
+		err = os.Chmod(filename, 0777)
+
+		if err != nil {
+			s.context.LogError(err)
+			return nil, engines.ErrNonFatalInternalError
+		}
+	}
+
+	env := make([]string, len(*s.env), len(*s.env)+len(environmentWhitelist))
```

Comment:
One of the things that we do currently in docker-worker that is being used in production is setting the folloowing environment variables https://github.com/taskcluster/docker-worker/blob/master/lib/task.js#L324-L329

I think these variables should be added to the sandbox's environment variables and cannot be overwritten by a plugin.